app/src/lib/stores/app-store.ts

```diff
@@ -2923,11 +2926,10 @@ export class AppStore extends TypedBaseStore<IAppState> {
         UncommittedChangesStrategyKind.moveToNewBranch &&
       checkoutSucceeded
     ) {
-      const stashToPop =
-        createdStashEntry || uncommittedChangesStrategy.transientStashEntry
+      stashToPop = stashToPop || uncommittedChangesStrategy.transientStashEntry
       if (stashToPop !== null) {
         await gitStore.performFailableOperation(() => {
-          return popStashEntry(repository, stashToPop.stashSha)
+          return popStashEntry(repository, stashToPop!.stashSha)
```

Comment:
This non-null assertion is required to appease the TS compiler.

Reply:
There's a slightly better way to appease the compiler without using the `!` here:

```diff
diff --git a/app/src/lib/stores/app-store.ts b/app/src/lib/stores/app-store.ts
index dd9c7460c..2127741c8 100644
--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -2928,8 +2928,9 @@ export class AppStore extends TypedBaseStore<IAppState> {
     ) {
       stashToPop = stashToPop || uncommittedChangesStrategy.transientStashEntry
       if (stashToPop !== null) {
+        const stashSha = stashToPop.stashSha
         await gitStore.performFailableOperation(() => {
-          return popStashEntry(repository, stashToPop!.stashSha)
+          return popStashEntry(repository, stashSha)
         })
       }
     }

```